### PR TITLE
chore(EMI-1446): add address autocomplete tracking

### PR DESCRIPTION
**Pull request by rquartararo**
The type of this PR is: **Chore**

This PR solves [EMI-1446]

### Description

This adds tracking for the address autocomplete according to https://github.com/artsy/cohesion/pull/457

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[EMI-1446]: https://artsyproduct.atlassian.net/browse/EMI-1446?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Components/Address/useAddressAutocomplete.ts [hunk 1]
```diff
@@ -154,15 +159,26 @@ export const useAddressAutocomplete = (
...
         setResult(finalSuggestions.slice(0, 5))
+
+        trackEvent({
+          action: ActionType.addressAutoCompletionResult,
+          context_module: ContextModule.ordersShipping,
+          context_page_owner_type: "orders-shipping",
+          context_page_owner_id: contextPageOwnerId,
+          input: search,
+          suggested_addresses_results: finalSuggestions.length,
+        })
```
Comment (erikdstock) [inline src/Components/Address/useAddressAutocomplete.ts:170]: nit: Can we add `as TheCohesionType` types to all of the params we pass to `trackEvent()`? This surfaces errors in our interfaces:
image

**Review by erikdstock — COMMENTED** [review]
Looks good with one minor comment to make sure we stick to the cohesion schema.

**Review by TMcMeans — DISMISSED** [review]
LGTM 👍🏾 🥇

File: src/Components/Address/useAddressAutocomplete.ts (hunk 1 above)
Comment (erikdstock) [reply src/Components/Address/useAddressAutocomplete.ts:170]: Update: In the case above i think our cohesion schema is actually inconsistent (see the doc comment vs the interface [here](https://github.com/artsy/cohesion/pull/457/files#diff-ee70df3a6cb0738382827014e641e2491d82c625fcd0596f7f1075da3454b724R249-R260)). I'd still like to add the `as TheType` bits to help us catch things like this.

Comment (TMcMeans) [reply src/Components/Address/useAddressAutocomplete.ts:170]: is the goal of adding` TheCohesionType` as a type to the params in trackEvent() to enforce type checking to make sure that the values compile as expected?

File: src/Components/Address/AddressForm.tsx [hunk 2]
```diff
@@ -112,6 +121,10 @@ export const AddressForm: React.FC<AddressFormProps> = ({
     }
     setKey(key)
     onChangeValue(key, ev.currentTarget.value)
+
+    if (userSelectedAddressOption && key !== "name" && key !== "phoneNumber") {
+      trackAutoCompleteEdit(key)
```
Comment (erikdstock) [inline src/Components/Address/AddressForm.tsx:126]: Sorry, one more nit here - can we only track this once per selected address? 
image

Comment (erikdstock) [reply src/Components/Address/AddressForm.tsx:126]: ah, maybe that makes sense - adding secondary info would mean something different than fixing the zip code. Your call - either track it the first time or once for each field, and we can confirm data's preference later (cc @daytavares)

File: src/Components/Address/AddressForm.tsx [hunk 3]
```diff
@@ -127,13 +123,13 @@ export const AddressForm: React.FC<AddressFormProps> = ({
     onChangeValue(key, ev.currentTarget.value)
 
     if (
-      !userEditedSelectedAddress &&
-      userSelectedAddressOption &&
+      selectedAddressOption.option &&
+      !selectedAddressOption.edited &&
```
Comment (erikdstock) [inline src/Components/Address/AddressForm.tsx]: nice :+1:

**Review by erikdstock — COMMENTED** [review]
Looking good - can we type the events as discussed in the [#web-practice slack thread here](https://artsy.slack.com/archives/C2TQ4PT8R/p1696363649011089?thread_ts=1696361425.671659&cid=C2TQ4PT8R)? Then ok to merge from me.

Edit: As i posted this i see the types were added. Thanks!

**Review by erikdstock — APPROVED** [review]
Thanks - Good to merge once we confirm these types with @daytavares!

File: src/Components/Address/useAddressAutocomplete.ts (hunk 1 above)
Comment (daytavares) [reply src/Components/Address/useAddressAutocomplete.ts]: Created the [PR](https://github.com/artsy/cohesion/pull/465)

**Pull request merged**